snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/SnomedRf2ExportRequest.java

```diff
@@ -346,80 +414,107 @@ public Rf2ExportResult execute(final RepositoryContext context) {
 			result.add(referenceBranch);
 			break;
 		}
-		return result.build();
+		
+		List<String> branchesToExport = ImmutableSortedSet.copyOf(result.build()).asList(); // eliminate duplicates
+		
+		Map<String, String> branchesAndRangesToExport = newHashMap();
+		
+		for (int i = 0; i < branchesToExport.size(); i++) {
+			
+			final String previousVersion = i == 0 ? null : branchesToExport.get(i - 1);
+			final String currentVersion = branchesToExport.get(i);
+
+			final String branchRangeToExport = previousVersion == null ? currentVersion : RevisionIndex.toRevisionRange(previousVersion, currentVersion);
+			
+			branchesAndRangesToExport.put(currentVersion, branchRangeToExport);
+			
+		}
+		
+		return branchesAndRangesToExport;
 	}
 
-	private String getArchiveEffectiveTime(final TreeSet<CodeSystemVersionEntry> versionsToExport, final Optional<CodeSystemVersionEntry> latestVersion) {
+	private Date getArchiveEffectiveTime(final TreeSet<CodeSystemVersionEntry> versionsToExport) {
 
+		Optional<CodeSystemVersionEntry> latestVersion = versionsToExport.isEmpty() ? Optional.empty() : Optional.of(versionsToExport.last());
+		
 		if (includePreReleaseContent) {
 
 			if (!transientEffectiveTime.isEmpty()) {
-				return transientEffectiveTime;
-			} else if (!versionsToExport.isEmpty()) {
-				return getNextEffectiveDate(versionsToExport.last().getEffectiveDate());
+				return adjustCurrentHour(Dates.parse(transientEffectiveTime, DateFormats.SHORT));
 			} else if (latestVersion.isPresent()) {
-				return getNextEffectiveDate(latestVersion.get().getEffectiveDate());
+				return adjustCurrentHour(getNextEffectiveDate(latestVersion.get().getEffectiveDate()));
 			}
 			
 		} else {
 			
-			if (!versionsToExport.isEmpty()) {
-				return EffectiveTimes.format(versionsToExport.last().getEffectiveDate(), DateFormats.SHORT);
-			} else if (latestVersion.isPresent()) {
-				return EffectiveTimes.format(latestVersion.get().getEffectiveDate(), DateFormats.SHORT);
+			if (latestVersion.isPresent()) {
+				return adjustCurrentHour(new Date(latestVersion.get().getEffectiveDate()));
 			}
+			
 		}
 		
-		return EffectiveTimes.format(Dates.todayGmt(), DateFormats.SHORT);
+		return adjustCurrentHour(Dates.parse(Dates.format(new Date(), TimeZone.getTimeZone("UTC"), DateFormats.DEFAULT)));
```

Comment:
The method name is technically incorrect; not all timezones have full hour differences. 
http://yourcalendricalfallacyis.com/#timezones-always-are-on-the-hour-mark 🤓